src/d3a/d3a_core/util.py

```diff
@@ -527,6 +537,25 @@ def convert_area_throughput_kVA_to_kWh(transfer_capacity_kWA, slot_length):
         if transfer_capacity_kWA is not None else 0.
 
 
+def find_timestamp_of_same_weekday_and_time(indict, time_slot, ignore_not_found=False):
+    if IS_CANARY_NETWORK:
+        start_time = list(indict.keys())[0]
```

Comment:
Are you sure that `indict ` is an ordered dict else I think its not guaranteed to be the first index as `start_time ` ?

Reply:
No, you are right. I mean it is ordered by the creation method, but i can not be sure and thus will use an OrderedDict (as long as this can oder by pendulum time stamp).